smart-contracts/contracts/mixins/MixinPurchase.sol

```diff
@@ -170,6 +158,44 @@ contract MixinPurchase is
     }
   }
 
+  /**
+  * @dev Extend function
+  * @param _value the number of tokens to pay for this purchase >= the current keyPrice - any applicable discount
+  * (_value is ignored when using ETH)
+  * @param _recipient address of the recipient of the key to extend
+  * @param _referrer address of the user making the referral
+  * @param _data arbitrary data populated by the front-end which initiated the sale
+  * @dev Throws if lock is disabled or key does not exist for _recipient. Throws if _recipient == address(0).
+  */
+  function extend(
+    uint256 _value,
+    address _recipient,
```

Comment:
Same as above. Maybe it is easier to already move toward tokenId for that function?

Reply:
The move to tokenId is on the other PR. It requires a complete rewriting of most tests and tons of changes in the contract, so very hard to isolate...